scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,172 @@
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int scopeStart;
+    public final Label[] end;
 
     /**
-     * The length of the local variable's scope (or live range), in bytes.
+     * The indices of the element being visited in the classfile.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int scopeLength;
+    public final int[] index;
 
-    public final String varName;
-    public final int varIndex;
+    /**
+     * The name of the local variable being visited.
+     *
+     * This is not part of the abstract state of the LocalLocation:
+     * it is not read by equals(), hashCode(), or toString().
+     */
+    public final @Nullable String variableName;
 
     /**
-     * Constructs a new {@link LocalLocation}; the arguments are assigned to
-     * the fields of the same names.
+     * Construct a new LocalLocation.  This constructor does not assign meaningful
+     * values to start or end.  Thus, the getScopeStart and getScopeLenth methods must not
+     * be used on the result.
+     *
+     * @param variableName the name of the local variable
+     * @param index the offset of the variable in the stack frame
      */
-    public LocalLocation(int index, int scopeStart, int scopeLength) {
+    public LocalLocation(String variableName, int index) {
+        this(new Label[] {new Label()}, new Label[] {new Label()}, new int[] {index}, variableName);
+    }
+
+    /**
+     * Construct a new LocalLocation.
+     *
+     * @param start the code offsets to the starts of the variable's lifetime(s)
+     * @param end the code offsets to the ends of the variable's lifetime(s)
+     * @param index the stack offsets of the variable's lifetime(s)
+     * @param variableName the name of the local variable
+     */
+    public LocalLocation(Label[] start, Label[] end, int[] index, String variableName) {
+        this.start = start;
+        this.end = end;
         this.index = index;
-        this.scopeStart = scopeStart;
-        this.scopeLength = scopeLength;
-        this.varName = null;
-        this.varIndex = -1;
+        this.variableName = variableName;
     }
 
-    public LocalLocation(String varName, int varIndex) {
-        this.index = -1;
-        this.scopeStart = -1;
-        this.scopeLength = -1;
-        this.varName = varName;
-        this.varIndex = varIndex;
+    /**
+     * Construct a new LocalLocation representing a single scope/lifetime.
+     * Only being used by Writers, not Readers for now. Should possibly deprecate this in the future.
+     * Changes values reflectively.
+     *
+     * @param index the offset of the variable in the stack frame
+     * @param scopeStart the bytecode offset of the start of the variable's lifetime
+     * @param scopeLength the bytecode length of the variable's lifetime
+     */
+    public LocalLocation(int index, int scopeStart, int scopeLength) {
```

Comment:
The order of arguments to this constructor is inconsistent with others, in that `index` comes first instead of after the start and length.  (Maybe index should come first because there is a constructor that takes it but not the start and length.  But in any event, make the constructors consistent.)